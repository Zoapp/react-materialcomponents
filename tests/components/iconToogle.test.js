import React from "react";
import renderer from "react-test-renderer";
import IconToggle from "@libs/components/iconToggle";

describe("components/IconToggle", () => {
  it("can be pressed", () => {
    const component = renderer.create(
      <IconToggle name="foo" nameOff="bar" pressed />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can be disabled", () => {
    const component = renderer.create(
      <IconToggle name="foo" nameOff="bar" disabled />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a label", () => {
    const component = renderer.create(
      <IconToggle name="foo" nameOff="bar" label="foobar" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a labelOff", () => {
    const component = renderer.create(
      <IconToggle name="foo" nameOff="bar" labelOff="foobar" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can be colored", () => {
    const component = renderer.create(
      <IconToggle name="foo" nameOff="bar" color="pink" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
