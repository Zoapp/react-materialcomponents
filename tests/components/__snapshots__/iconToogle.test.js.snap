// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IconToggle can be colored 1`] = `
<i
  aria-pressed={false}
  arria-label={null}
  className="mdc-icon-toggle material-icons"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  style={
    Object {
      "color": "pink",
    }
  }
  tabIndex="0"
>
  bar
</i>
`;

exports[`components/IconToggle can be disabled 1`] = `
<i
  aria-pressed={false}
  arria-label={null}
  className="mdc-icon-toggle material-icons mdc-icon-toggle--disabled"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  style={Object {}}
  tabIndex="0"
>
  bar
</i>
`;

exports[`components/IconToggle can be pressed 1`] = `
<i
  aria-pressed={true}
  arria-label={null}
  className="mdc-icon-toggle material-icons"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  style={Object {}}
  tabIndex="0"
>
  foo
</i>
`;

exports[`components/IconToggle can have a label 1`] = `
<i
  aria-pressed={false}
  arria-label={null}
  className="mdc-icon-toggle material-icons"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  style={Object {}}
  tabIndex="0"
>
  bar
</i>
`;

exports[`components/IconToggle can have a labelOff 1`] = `
<i
  aria-pressed={false}
  arria-label="foobar"
  className="mdc-icon-toggle material-icons"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  style={Object {}}
  tabIndex="0"
>
  bar
</i>
`;
